core/chain_manager.go

```diff
@@ -335,23 +319,23 @@ func (self *ChainManager) ExportN(w io.Writer, first uint64, last uint64) error
 // insert injects a block into the current chain block chain. Note, this function
 // assumes that the `mu` mutex is held!
 func (bc *ChainManager) insert(block *types.Block) {
-	err := WriteHead(bc.chainDb, block)
-	if err != nil {
-		glog.Fatal("db write fail:", err)
+	// Insert the block as the new head of the chain
```

Comment:
No, canonical hash simply assigns a hash to a number. It cannot do more, as during every reorg it needs to update the entire new canonical subchain's numbers, not just the heads. The head is set separately via `WriteHeadBlockHash`.